downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/HumanPathwaysWithDiagrams.java

```diff
@@ -38,19 +47,57 @@ public static void execute(MySQLAdaptor dba, String releaseNumber) throws Except
             // If returned value from 'representedPathway' referral is not null, it is considered to have a pathway diagram
+
             if (representedPathwayInstances != null) {
-                // Generate line in file and append to that file
-                String line = humanPathway.getDBID() + "\t" + humanPathway.getDisplayName() + "\t" + isDisease + "\n";
-                Files.write(Paths.get(humanPathwaysWithDiagramsFilename), line.getBytes(), StandardOpenOption.APPEND);
+                if (!onlyGreenBoxDiagrams(representedPathwayInstances)) {
+                    // Generate line in file and append to that file
+                    String line = humanPathway.getDBID() + "\t" + humanPathway.getDisplayName() + "\t" + isDisease + "\n";
+                    Files.write(Paths.get(humanPathwaysWithDiagramsFilename), line.getBytes(), StandardOpenOption.APPEND);
+                }
             }
         }
         // Move file to directory pertaining to release
         String outpathName = releaseNumber + "/" + humanPathwaysWithDiagramsFilename;
         Files.move(Paths.get(humanPathwaysWithDiagramsFilename), Paths.get(outpathName), StandardCopyOption.REPLACE_EXISTING);
     }
+
+    // Method for checking if diagrams are only comprised of GreenBoxDiagrams/Compartments, which we want to exclude
+    // Involves going through the PathwayDiagram XML and searching for particular strings that denote the green boxes and compartments.
+    private static boolean onlyGreenBoxDiagrams(Collection<GKInstance> representedPathwayInstances) throws Exception {
+        DocumentBuilderFactory documentBuildFactory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder documentBuilder = documentBuildFactory.newDocumentBuilder();
+
+        for (GKInstance pathwayDiagramInst : representedPathwayInstances) {
+            String pathwayDiagramXML = pathwayDiagramInst.getAttributeValue(ReactomeJavaConstants.storedATXML).toString();
+            Document parsedXMLDoc = documentBuilder.parse(new InputSource(new StringReader(pathwayDiagramXML)));
+            parsedXMLDoc.getDocumentElement().normalize();
+            NodeList nList = parsedXMLDoc.getElementsByTagName("Nodes");
+            for (int n = 0; n < nList.getLength(); n++) {
+                Node node = nList.item(n);
+                NodeList nChildList = node.getChildNodes();
+                for (int c = 0; c < nChildList.getLength(); c++) {
+                    Node childNode = nChildList.item(c);
+                    if (childNode.getNodeType() == Node.ELEMENT_NODE) {
+                        if (childNode.getNodeName() != PROCESS_NODE_STRING) {
+                            if (childNode.getNodeName() != RENDERABLE_COMPARTMENT_STRING) {
+                                return false;
+                            }
+                        }
+                    }
```

Comment:
Could these also be joined with `&&` rather than nesting?

Reply:
You could even name the condition in a small private method if you like for readability:

`private static boolean isNotAPathwayOrCompartmentElement(childNode)`